fluent-package/templates/package-scripts/fluent-package/deb/postinst

```diff
@@ -22,38 +20,113 @@ add_system_user() {
     fi
 }
 
+v4migration=/tmp/.<%= project_name %>_v4migration
+v4migration_with_restart=/tmp/.<%= project_name %>_v4migration_with_restart
+
 add_directories() {
     mkdir -p /var/run/<%= package_dir %>
-    mkdir -p /etc/<%= compat_package_dir %>
-    mkdir -p /etc/<%= compat_package_dir %>/plugin
-    mkdir -p /var/log/<%= compat_package_dir %>
+    mkdir -p /etc/<%= package_dir %>
+    mkdir -p /etc/<%= package_dir %>/plugin
+    mkdir -p /var/log/<%= package_dir %>
+    if [ -d /etc/<%= compat_package_dir %> -a ! -h /etc/<%= compat_package_dir %> ]; then
+	touch $v4migration
```

Comment:
Why don't we use `prevver` to judge if we need migration or not?

Reply:
prevver is not appropriate in this case because it is empty, if it is upgrade from same package, it is reasonable though.

Reply:
Oh, I see. Thanks!